drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -29,39 +29,25 @@ local COLOR_TEMPERATURE_KELVIN_MIN = 1
 local COLOR_TEMPERATURE_MIRED_MAX = CONVERSION_CONSTANT/COLOR_TEMPERATURE_KELVIN_MIN
 local COLOR_TEMPERATURE_MIRED_MIN = CONVERSION_CONSTANT/COLOR_TEMPERATURE_KELVIN_MAX
 
+local ENDPOINT_TO_COMPONENT_MAP = "__endpoint_to_component_map"
```

Comment:
Thoughts on checking for the existence of this field on each device on driver init and populating the map if it doesn't exist? The goal would be to fix devices that are already joined without requiring them to be removed and re-commissioned.

Reply:
That would be a great way of dealing with existing devices, and both, device_init and device_add could call into a commonly share function, if the map is null.

Reply:
Also, if this could be implemented, it would be helpful to log the creation of the map, especially for existing devices.

Reply:
I think this is a great idea, Tom! Although on a side note, I am wondering if there is any precedent in the past for reconfiguring already joined devices? Were we just supposed to remove and rejoin the devices? @greens  do you know if there is any precedent?

Reply:
I know we have done fixes in device init for devices that aren't functioning properly due to bugs in the way we add them. The risk of the change is heightened though, because the change then impacts all matter devices using the switch driver rather than just newly joined devices.

Reply:
In an attempt to reduce complexity of dealing with devices in the field that have a ep to comp map and ones that dont, we should probably do reprocessing on init of the ep/comp map. It will help us ensure correctness with unit tests.

Reply:
Moved component mapping to `init` and removed `added` in a9e0c3ac0e37798c0fd3f4927ce6b4ad80d88fc0